### PR TITLE
ROC-7975 Set the current time as the date referred for direction

### PR DESCRIPTION
**Pull request by AlokDatta**
When mediation is unsuccessful, the date referred to "Ready for directions - Judge"  is shown as the date that mediation was pending instead of the date when the mediation was set as unsuccessful by the caseworker. This needs to be corrected.

JIRA link (if applicable)

https://tools.hmcts.net/jira/browse/ROC-7975

Change description

Modified the MediationFailedCallbackHandler.java to reflect the current date time as the date referred for directions..

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by andrewwa-kainos** [comment]
Please add/update a unit test to verify this change

**Comment by AlokDatta** [comment]
> Please add appropriate tests

Hello Irfan/Andrew,

 I have Added a new test case, can you please have a look.

Thanks,
Alok

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandlerTest.java [hunk 1]
```diff
@@ -294,6 +310,29 @@ public void shouldTriggerDQDeadlineEventIfOfflineDQ() {
                 .saveCaseEvent(AUTHORISATION, claim.getCcdCaseId(), CaseEvent.DIRECTIONS_QUESTIONNAIRE_DEADLINE);
     }
 
+    @Test
+    public void testReadyForDirectionsDateForMediationUnsuccessful() {
+        clock = Clock.fixed(Instant.parse(INSTANT_EXPECTED), ZoneId.of("UTC"));
+        LocalDateTime dateTime = LocalDateTime.now(clock);
```
Comment (andrewwa-kainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandlerTest.java:316]: No point making a `@Mock` clock and then overwriting it before ever using it. Also as this is just a reference instant you don't need a clock.
```suggestion
        var dateTime = LocalDateTime.parse(INSTANT_EXPECTED);
```

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandlerTest.java [hunk 2]
```diff
@@ -309,4 +348,18 @@ private void handleSubmittedCallback() {
...
+        return CCDCase.builder()
+            .dateReferredForDirections(dateTime)
+            .build();
+    }
+
+    private Map<String, Object> getCcdCaseMap(LocalDateTime dateTime) {
+        Map<String, Object> map = new HashMap<>();
+        map.put(DATE_REFERRED_FOR_DIRECTIONS, getCcdCase(dateTime));
+        map.put(REFERENCE_KEY, REFERENCE);
+        return map;
```
[the remark this answers is not in the document]
Comment (AlokDatta) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandlerTest.java:357]: We need a mutable map here, as there are operations in the code to put the 'STATE' as an entry. That's why kept this as is.

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandlerTest.java [hunk 3]
```diff
@@ -294,6 +310,29 @@ public void shouldTriggerDQDeadlineEventIfOfflineDQ() {
...
+        when(caseMapper.to(any(Claim.class))).thenReturn(getCcdCase(dateTime));
+        when(caseDetailsConverter.convertToMap(any(CCDCase.class))).thenReturn(getCcdCaseMap(dateTime));
+
+        AboutToStartOrSubmitCallbackResponse response = (AboutToStartOrSubmitCallbackResponse)
+            mediationFailedCallbackHandler
+                .handle(callbackParams);
+
+        CCDCase ccdCase = (CCDCase) response.getData().get("dateReferredForDirections");
+
+        assertEquals(ccdCase.getDateReferredForDirections(), dateTime);
```
[the remark this answers is not in the document]
Comment (AlokDatta) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandlerTest.java]: Suggestion taken.

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandlerTest.java [hunk 4]
```diff
@@ -294,6 +310,29 @@ public void shouldTriggerDQDeadlineEventIfOfflineDQ() {
...
+
+        when(caseDetailsConverter.extractClaim(any(CaseDetails.class))).thenReturn(claim);
+        when(caseMapper.to(any(Claim.class))).thenReturn(getCcdCase(dateTime));
+        when(caseDetailsConverter.convertToMap(any(CCDCase.class))).thenReturn(getCcdCaseMap(dateTime));
+
+        AboutToStartOrSubmitCallbackResponse response = (AboutToStartOrSubmitCallbackResponse)
+            mediationFailedCallbackHandler
+                .handle(callbackParams);
+
+        CCDCase ccdCase = (CCDCase) response.getData().get("dateReferredForDirections");
```
[the remark this answers is not in the document]
Comment (AlokDatta) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandlerTest.java]: Suggestion taken.

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandlerTest.java [hunk 5]
```diff
@@ -294,6 +310,29 @@ public void shouldTriggerDQDeadlineEventIfOfflineDQ() {
...
+            .response(
+                SampleResponse.FullDefence.builder().build()).build();
+
+        when(caseDetailsConverter.extractClaim(any(CaseDetails.class))).thenReturn(claim);
+        when(caseMapper.to(any(Claim.class))).thenReturn(getCcdCase(dateTime));
+        when(caseDetailsConverter.convertToMap(any(CCDCase.class))).thenReturn(getCcdCaseMap(dateTime));
+
+        AboutToStartOrSubmitCallbackResponse response = (AboutToStartOrSubmitCallbackResponse)
+            mediationFailedCallbackHandler
+                .handle(callbackParams);
```
[the remark this answers is not in the document]
Comment (AlokDatta) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandlerTest.java]: Suggestion taken.

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandlerTest.java (hunk 1 above)
Comment (AlokDatta) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandlerTest.java]: Suggestion taken.

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandlerTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (andrewwa-kainos) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandlerTest.java:357]: okay, Sure!

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandlerTest.java (hunk 1 above)
Comment (andrewwa-kainos) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandlerTest.java]: Thanks! Also watch out for unused imports

Comment (AlokDatta) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandlerTest.java]: Checked. Thanks.

**Comment by AlokDatta** [comment]
Closing this pull request as build is failing in Jenkins

**Pull request closed without merging**